Termite enemy

desultisoft opened the pull request:
Created a termite enemy tscn that explodes when close to the player.
It also can die and chase the player.
It uses architecture fairly similar to the spiker.

desultisoft committed:
Made termite

desultisoft committed:
I finished an iteration of the Termite

It walks up to the player and explodes.
It uses code pretty similar to the spiker.
I ended up using an animation player for it as it simplified things.
It creates an explosion scene that cleans itself up when it finishes the attack.

Flamebeard10339 commented on src/Code/Enemies/Termite/Termite.gd:
```diff
@@ -0,0 +1,80 @@
+		target = null
+		return
+
+	if not attacking and !dead:
```
not or !. One or the other

Flamebeard10339 commented on src/project.godot:
```diff
@@ -95,6 +95,10 @@ window/size/test_height=540
+
+enabled=PoolStringArray(  )
+
 [importer_defaults]
```
I try to avoid adding the weird random changes that godot adds to github. This change fyi is because you enabled/disabled a plugin. but it doesn't do anything other than mess with merges

desultisoft replied:
Ok. I will try to avoid this in the future.

Flamebeard10339 commented on src/Code/Player/Player.tscn:
```diff
@@ -77,4 +77,5 @@ script = ExtResource( 4 )
 [node name="IFramesTimer" type="Timer" parent="."]
 wait_time = 0.35
 one_shot = true
+
```
see the project.godot comment

Flamebeard10339 commented on src/Code/Enemies/Enemy.tscn:
```diff
@@ -31,4 +31,5 @@ __meta__ = {
 }
 
 [node name="bleeding_debuffs" type="Node" parent="."]
+
```
see the project.godot comment

Flamebeard10339 commented on src/Code/Enemies/Termite/Termite.gd:
```diff
@@ -0,0 +1,80 @@
+		dead = true
+		frozen = true
+		move_dir = Vector2.ZERO
+		display.play("Die")
```
This does not free the node. If you are trying to leave behind a corpse then make sure to disable the collision shape so that the player can walk over the corpse.

Flamebeard10339 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/7536002/141015285-8e1bee0c-dd03-4b12-b831-685d16629c0d.png)

You have a gap in the Run animation making him stutter.  Die also has a gap. 

The termite needs to queue_free() after the die method. It breaks gratuitous violence if the enemy dies more than once.

desultisoft committed:
Termite now uses Sprite Frames and timer method for consistency sake with the rest of the game.

desultisoft committed:
Revert "Termite now uses Sprite Frames and timer method for consistency sake with the rest of the game."

This reverts commit 8e326cb17b86368645eb7e5ab8c749311bfbfa5b.

desultisoft committed:
I cleaned up some termite related problems.

Added some simple code that flips the termite.
I fixed the merge conflict with main in world.

desultisoft committed:
Fixed onhit effects breaking the game.

Changed some lines in enemy death code and termite death code in order to make
things not break. Not sure if I have removed the effects on death however.

OceanBoy612 committed:
Added a testing world to game
